### ISSUE TITLE
allow parameters to be marked as mutable

### ISSUE THREAD

**Issue by spotandjake** [grain-lang/grain#1964]
allow function paramters to be marked as mutable syntax example being 
```
let t = (mut str) => {
  str += 1
  str
}
```
The semantics of this would be equivlant to 
```
let t = (str) => {
  let mut str = str
  str += 1
  str
}
```

**Comment by alex-snezhko** [comment]
I feel like this may be confusing as someone may expect this to mutate the value that you are actually passing in. There's even a whole blog post I found for Rust with someone having this same confusion with this feature https://www.snoyman.com/blog/2020/05/no-mutable-parameters-in-rust/

**Comment by spotandjake** [comment]
> I feel like this may be confusing as someone may expect this to mutate the value that you are actually passing in. There's even a whole blog post I found for Rust with someone having this same confusion with this feature https://www.snoyman.com/blog/2020/05/no-mutable-parameters-in-rust/

Reading over that article it seems a lot of the confusion in rust comes from the `&mut` and mutable values, currently in grain `mut` indicates we are mutating the variable not the value, which I think solves the confusion of the feature as a whole.

**Comment by alex-snezhko** [comment]
I agree that once you understand that's what `mut` means in Grain you wouldn't make that mistake, but my concern was that a Grain novice may be tripped up by this, in the same way that the Rust novice from the perspective of the article may be upon seeing `mut` and knowing that this means "this thing is mutable".

**Comment by spotandjake** [comment]
Thats valid but I feel like its not a confusing concept itself, docs would solve the issue and the feature matches the semantics of the language. Additionally we already have other things that could be confusing like this. Arguably the fact that you can do:
```rs
let arr = [> 1, 2]
arr[0] += 5
```
could be confusing as you are not marking it as mutable.
